# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/xul/MenuBarListener.cpp
# Commit: 6c4655ca0e97
# Full Hash: 6c4655ca0e978dc82bb5686979698790412ff2da
# Author: Magnus Melin <mkmelin+mozilla@iki.fi>
# Date: 2023-03-02 04:57:23
# Description:
#   Bug 1817649 - Fix crash in [@ mozilla::dom::EventTarget::ComputeWantsUntrusted]. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171296
# ==============================================================================

diff -r f883d6d51f1a -r 6c4655ca0e97 dom/xul/MenuBarListener.cpp
--- a/dom/xul/MenuBarListener.cpp	Wed Mar 01 17:59:03 2023 +0000
+++ b/dom/xul/MenuBarListener.cpp	Wed Mar 01 18:26:11 2023 +0000
@@ -67,7 +67,9 @@
 
   // Needs to listen to the deactivate event of the window.
   RefPtr<EventTarget> top = nsContentUtils::GetWindowRoot(mEventTarget);
-  top->AddSystemEventListener(u"deactivate"_ns, this, true);
+  if (!NS_WARN_IF(!top)) {
+    top->AddSystemEventListener(u"deactivate"_ns, this, true);
+  }
 }
 
 ////////////////////////////////////////////////////////////////////////
